SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -0,0 +1,243 @@
+
+			if (IsValid(PlayerController))
+			{
+				ClientOneSpawnedPawn =
+					GetWorld()->SpawnActor<ATestMovementCharacter>(CharacterSpawnLocation, FRotator::ZeroRotator, FActorSpawnParameters());
+				RegisterAutoDestroyActor(ClientOneSpawnedPawn);
+
+				ClientOneDefaultPawn = PlayerController->GetPawn();
+
+				PlayerController->Possess(ClientOneSpawnedPawn);
+
+				FinishStep();
+			}
+		});
+	}
+
+	{ // Step 1 - All workers check if they have one AReplicatedGASTestActor in the world, and set a reference to it.
+		AddStep(
+			TEXT("DynamicSubobjectsTestAllWorkers"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				TArray<AActor*> FoundActors;
+				UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
+
+				if (FoundActors.Num() == 1)
+				{
+					TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
+
+					if (IsValid(TestActor))
+					{
+						FinishStep();
+					}
+				}
+			},
+			StepTimeLimit);
+	}
+
+	{ // Step 2 - Client 1 checks if it has correctly possessed the TestMovementCharacter.
+		AddStep(
+			TEXT("DynamicSubobjectsTestClientCheckPossesion"), FWorkerDefinition::Client(1), nullptr, nullptr,
+			[this](float DeltaTime) {
+				ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+				APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+				ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+				if (IsValid(PlayerController))
+				{
+					if (PlayerCharacter == PlayerController->AcknowledgedPawn)
+					{
+						FinishStep();
+					}
+				}
+			},
+			StepTimeLimit);
+	}
+
+	for (int i = 0; i < DynamicComponentsPerClass + 1; ++i)
+	{
+		{ // Step 3 - Server moves the TestMovementCharacter of Client 1 to a remote location, so that it does not see the
+		  // AReplicatedGASTestActor.
+			AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
+				if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation))
+				{
+					if (ClientOneSpawnedPawn->GetActorLocation().Equals(CharacterRemoteLocation, 1.0f))
+					{
+						FinishStep();
+					}
+				}
+			});
+		}
+
+		{ // Step 4 - Client 1 makes sure that the movement was correctly replicated
+			AddStep(
+				TEXT("DynamicSubobjectsTestClientCheckFirstMovement"), FWorkerDefinition::Client(1), nullptr, nullptr,
+				[this](float DeltaTime) {
+					ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
+					APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
+					ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
+
+					if (IsValid(PlayerCharacter))
+					{
+						if (PlayerCharacter->GetActorLocation().Equals(CharacterRemoteLocation, 50.0f))
```

Comment:
Mostly a curiosity thing, but do you need such a big tolerance in the equal?

Reply:
Oh thats left there from the time we had issues to move the character in tests I am going to decrease it.